src/uwtools/config/formats/base.py

```diff
@@ -44,14 +43,21 @@ def __init__(self, config: Optional[Union[dict, Path]] = None) -> None:
 
     def __repr__(self) -> str:
         """
-        Returns the YAML string representation of a Config object.
+        Returns the string representation of a Config object.
         """
-        s = StringIO()
-        yaml.dump(self.data, s)
-        return s.getvalue()
+        return self._dict_to_str(self.data)
```

Comment:
Previously, this method -- inherited by all subclasses -- represented the config as YAML, regardless of its actual format. Now that all `Config` subclasses implement a `_dict_to_str()` method (see below) and can represent themselves in their own format, the inherited `__repr__()` method can simply call that, delegating to each subclass.